src/EprPrnIntegration.Api/UpdateProducersFunction.cs

```diff
@@ -48,22 +50,31 @@ public async Task Run([TimerTrigger("%UpdateProducersTrigger%")] TimerInfo myTim
         }
 
         var npwdUpdatedProducers = ProducerMapper.Map(updatedEprProducers, configuration);
-        var pEprApiResponse = await npwdClient.Patch(npwdUpdatedProducers, NpwdApiPath.UpdateProducers);
 
-        if (pEprApiResponse.IsSuccessStatusCode)
+        try
         {
-            logger.LogInformation(
-                "Producers list successfully updated in NPWD for time period {FromDate} to {ToDate}.", fromDate,
-                toDate);
+            var pEprApiResponse = await npwdClient.Patch(npwdUpdatedProducers, NpwdApiPath.UpdateProducers);
 
-            await utilities.SetDeltaSyncExecution(deltaRun, toDate);
+            if (pEprApiResponse.IsSuccessStatusCode)
+            {
+                logger.LogInformation(
+                    "Producers list successfully updated in NPWD for time period {FromDate} to {ToDate}.", fromDate,
+                    toDate);
+
+                await utilities.SetDeltaSyncExecution(deltaRun, toDate);
+            }
+            else
+            {
+                var responseBody = await pEprApiResponse.Content.ReadAsStringAsync();
+                logger.LogError(
+                    "Failed to update producer lists. error code {StatusCode} and raw response body: {ResponseBody}",
+                    pEprApiResponse.StatusCode, responseBody);
+            }
         }
-        else
+        catch (Exception ex)
         {
-            var responseBody = await pEprApiResponse.Content.ReadAsStringAsync();
-            logger.LogError(
-                "Failed to update producer lists. error code {StatusCode} and raw response body: {ResponseBody}",
-                pEprApiResponse.StatusCode, responseBody);
+            logger.LogError(ex, $"An error was encountered on attempting to call NPWD API {NpwdApiPath.UpdateProducers}");
```

Comment:
Looking into above logic I feel 429 will end up on above else statement can we please confirm that's not the case

Reply:
done. checked